fireworks_poe_bot/fw_poe_text_bot.py

```diff
@@ -473,42 +473,157 @@ def merge_messages_groups(
             generated_len = 0
             complete_response = ""
             unreplaced_complete_response = ""
-            async for response in self.completion_async_method(
-                model=self.model,
-                messages=messages,
-                stream=True,
-                request_timeout=self.request_timeout or 600,
-                temperature=query.temperature if query.temperature is not None else 0.6,
-                stop=stop_seqs,
-                max_tokens=self.max_tokens,
-                prompt_truncate_len=self.prompt_truncate_len,
-                **additional_args,
-            ):
-                # Step 3: Transform the CompletionStreamResponse into PartialResponse format
-                for choice in response.choices:
-                    assert isinstance(choice, ChatCompletionResponseStreamChoice)
-                    if choice.delta.content is None:
-                        continue
-
-                    if self.replace_think:
-                        unreplaced_complete_response += choice.delta.content
-                        thinking_mode = False
-                        if '<think>' in unreplaced_complete_response:
-                            choice.delta.content = choice.delta.content.replace('<think>', 'Thinking...\n')
-                            thinking_mode = True
-                        if '</think>' in unreplaced_complete_response:
-                            choice.delta.content = choice.delta.content.replace('</think>', '\n')
+            
+            # Add timestamps for streaming diagnostics
+            stream_start_time = time.time()
+            first_token_received = False
+            last_token_time = stream_start_time
+            token_count = 0
+            
+            self._log_info({
```

Comment:
we should have some request_id to correlate it between multiple messages. We can also pass it as `x-request-id` header when calling completions API so we can find it on server side